sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/expressions.scala

```diff
@@ -441,6 +456,24 @@ object BooleanSimplification extends Rule[LogicalPlan] with PredicateHelper {
 
       case Not(IsNull(e)) => IsNotNull(e)
       case Not(IsNotNull(e)) => IsNull(e)
+
+      // Using Not(null) == null rules
+      case IsNull(Not(e)) => IsNull(e)
+      case IsNotNull(Not(e)) => IsNotNull(e)
+
+      // Using (Not(a) === b) == (a === Not(b)), (Not(a) <=> b) == (a <=> Not(b)) rules
+      case EqualTo(Not(a), b) if canSimplifyNot(b) => EqualTo(a, Not(b))
```

Comment:
why is `EqualTo(a, Not(b))` better than `EqualTo(Not(a), b)`?

Reply:
For example of the `EqualTo(Not(a), b)` where `b = Not(c)`, it will become `EqualTo(a, Not(b))` => `EqualTo(a, Not(Not(c)))` => `EqualTo(a, c)`
In addition, `if canSimplifyNot(b)` checks if the optimization can converge that avoids the situation two conditions are returning to each other.
I plan to add those explanations to the comment.